src/api/tokenList/TokenListApi.ts

```diff
@@ -13,21 +26,67 @@ export interface TokenList {
 export class TokenListApiImpl implements TokenList {
   public networkIds: number[]
   private _tokensByNetwork: { [networkId: number]: TokenDetails[] }
+  private _tokenAddressNetworkSet: Set<string>
 
   public constructor(networkIds: number[]) {
     this.networkIds = networkIds
 
     // Init the tokens by network
     this._tokensByNetwork = {}
+    this._tokenAddressNetworkSet = new Set<string>()
+
     networkIds.forEach(networkId => {
       // initial value
-      this._tokensByNetwork[networkId] = getTokensByNetwork(networkId)
+      const tokenList = getTokensByNetwork(networkId).concat(this.loadUserTokenList(networkId))
```

Comment:
This is something I'm dealing with in my dynamic tokens PR, so you may not need to change anything.
Just raising the point that, we can have duplicates here. If a token added locally is later added to list of tokens.

Reply:
that is an interesting point
not gonna do anything about it, though :stuck_out_tongue: 
not here